exercises/affine-cipher/affine_cipher_test.rb

```diff
@@ -1,122 +1,149 @@
+  def test_encode_omg
     skip
-    cipher = Affine.new
-    cipher.addkey(21, 3)
-    assert_equal 'lvz', cipher.encode('OMG')
+    cipher = Affine.new(21, 3)
+    plaintext = 'OMG'
+    ciphertext = 'lvz'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
-  def test_encode_O_M_G
+  def test_encode_o_m_g
     skip
-    cipher = Affine.new
-    cipher.addkey(25, 47)
-    assert_equal 'hjp', cipher.encode('O M G')
+    cipher = Affine.new(25, 47)
+    plaintext = 'O M G'
+    ciphertext = 'hjp'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_mindblowingly
     skip
-    cipher = Affine.new
-    cipher.addkey(11, 15)
-    assert_equal 'rzcwa gnxzc dgt', cipher.encode('mindblowingly')
+    cipher = Affine.new(11, 15)
+    plaintext = 'mindblowingly'
+    ciphertext = 'rzcwa gnxzc dgt'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_numbers
     skip
-    cipher= Affine.new
-    cipher.addkey(3, 4)
-    assert_equal 'jqgjc rw123 jqgjc rw', cipher.encode('Testing,1 2 3, testing')
+    cipher = Affine.new(3, 4)
+    plaintext = 'Testing,1 2 3, testing.'
+    ciphertext = 'jqgjc rw123 jqgjc rw'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_deep_thought
     skip
-    cipher = Affine.new
-    cipher.addkey(5, 17)
-    assert_equal 'iynia fdqfb ifje', cipher.encode('Truth is fiction.')
+    cipher = Affine.new(5, 17)
+    plaintext = 'Truth is fiction.'
+    ciphertext = 'iynia fdqfb ifje'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_all_the_letters
     skip
-    cipher = Affine.new
-    cipher.addkey(17, 33)
+    cipher = Affine.new(17, 33)
     plaintext = 'The quick brown fox jumps over the lazy dog.'
     ciphertext = 'swxtj npvyk lruol iejdc blaxk swxmh qzglf'
     assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_with_a_not_coprime_to_m
     skip
-    cipher = Affine.new
-    assert_raises(ArgumentError) { cipher.addkey(6, 17) }
+    assert_raises(ArgumentError) { Affine.new(6, 17) }
   end
 
   def test_decode_exercism
     skip
-    cipher = Affine.new
-    cipher.addkey(3, 7)
-    assert_equal 'exercism', cipher.decode('tytgn fjr')
+    cipher = Affine.new(3, 7)
+    plaintext = 'exercism'
+    ciphertext = 'tytgn fjr'
+    assert_equal plaintext, cipher.decode(ciphertext)
   end
 
   def test_decode_a_sentence
     skip
-    cipher = Affine.new
-    cipher.addkey(19, 16)
+    cipher = Affine.new(19, 16)
     plaintext = 'anobstacleisoftenasteppingstone'
     ciphertext = 'qdwju nqcro muwhn odqun oppmd aunwd o'
     assert_equal plaintext, cipher.decode(ciphertext)
   end
 
   def test_decode_numbers
     skip
-    cipher = Affine.new
-    cipher.addkey(25, 7)
-    assert_equal 'testing123testing', cipher.decode('odpoz ub123 odpoz ub')
+    cipher = Affine.new(25, 7)
+    plaintext = 'testing123testing'
+    ciphertext = 'odpoz ub123 odpoz ub'
```

Comment:
I wonder if these should be the other way around in the decode tests.
`ciphertext` first and `plaintext` second?

Reply:
I thought about doing this before. I might try it out.